detection_rules/version_lock.py

```diff
@@ -286,14 +302,6 @@ def log_changes(r, route_taken, new_rule_version, *msg):
                 else:
                     raise RuntimeError("Unreachable code")
 
-                if 'previous' in lock_from_file:
```

Comment:
While this code worked, the flaw was that it relied on the root (current) version as a comparison. This meant that when the root version incremented, it would quietly allow previous entries to creep as well. 

L290-295 instead use the newly pinned `max_allowable_version` to keep a static check